apiserver/facades/client/spaces/spaces.go

```diff
@@ -427,6 +505,19 @@ func (api *API) checkSupportsSpaces() error {
 	return nil
 }
 
+// checkSupportsProviderSpaces checks if the environment implements NetworkingEnviron
+// and also if it support provider spaces.
+func (api *API) checkSupportsProviderSpaces() error {
+	env, err := environs.GetEnviron(api.backing, environs.New)
+	if err != nil {
+		return errors.Annotate(err, "getting environ")
+	}
+	if !environs.SupportsProviderSpaces(api.context, env) {
+		return errors.NotSupportedf("provider spaces")
```

Comment:
This will show the message "provider spaces not supported" to the operator. A better message will be "renaming provider-sourced spaces not supported".